docs: update README and tidy up docs (#619)
openebs/zfs-localpv: README.md

[![Community Meetings](https://img.shields.io/badge/Community-Meetings-blue)](https://us05web.zoom.us/j/87535654586?pwd=CigbXigJPn38USc6Vuzt7qSVFoO79X.1)
[![Go Report](https://goreportcard.com/badge/github.com/openebs/zfs-localpv)](https://goreportcard.com/report/github.com/openebs/zfs-localpv)

## Overview

### What is OpenEBS ZFS LocalPV?
OpenEBS ZFS LocalPV is a [CSI](https://github.com/container-storage-interface/spec) plugin for implementation of [ZFS](https://en.wikipedia.org/wiki/ZFS) backed persistent volumes for Kubernetes. It is a local storage solution, which means the device, volume and the application are on the same host. It doesn't contain any dataplane, i.e only its simply a control-plane for the kernel zfs volumes. It mainly comprises of two components which are implemented in accordance to the CSI Specs:

1. CSI Controller - Frontends the incoming requests and initiates the operation.
2. CSI Node Plugin - Serves the requests by performing the operations and making the volume available for the initiator.

### Why OpenEBS ZFS LocalPV?

> 1. Light weight, easy to setup storage provisoner for provisioning node local volumes in k8s ecosystem.

Reviewer: ```suggestion
1. Lightweight, easy to set up storage provisoner for host-local volumes in k8s ecosystem.
```

[![Community Meetings](https://img.shields.io/badge/Community-Meetings-blue)](https://us05web.zoom.us/j/87535654586?pwd=CigbXigJPn38USc6Vuzt7qSVFoO79X.1)
[![Go Report](https://goreportcard.com/badge/github.com/openebs/zfs-localpv)](https://goreportcard.com/report/github.com/openebs/zfs-localpv)

## Overview

> ### What is OpenEBS ZFS LocalPV?

Reviewer 2: ```suggestion
### What is OpenEBS ZFS LocalPV?

```

OpenEBS ZFS LocalPV is a [CSI](https://github.com/container-storage-interface/spec) plugin for implementation of [ZFS](https://en.wikipedia.org/wiki/ZFS) backed persistent volumes for Kubernetes. It is a local storage solution, which means the device, volume and the application are on the same host. It doesn't contain any dataplane, i.e only its simply a control-plane for the kernel zfs volumes. It mainly comprises of two components which are implemented in accordance to the CSI Specs:

1. CSI Controller - Frontends the incoming requests and initiates the operation.
2. CSI Node Plugin - Serves the requests by performing the operations and making the volume available for the initiator.

> ### Why OpenEBS ZFS LocalPV?

Reviewer 2: ```suggestion
### Why OpenEBS ZFS LocalPV?

```

1. Lightweight, easy to set up storage provisoner for host-local volumes in k8s ecosystem.

> 2. Makes ZFS stack available to k8s, allowing end users to use the ZFS functionalites like snapshot, restore, clone, thin provisioning, resize, encryption, compression, dedup, etc for their Persistent Volumes.

Reviewer 2: ```suggestion
2. Makes ZFS stack available to K8s, allowing end users to use the ZFS functionalites like snapshot, restore, clone, thin provisioning, resize, encryption, compression, dedup, etc for their Persistent Volumes.
```

3. Cloud native, i.e based on CSI spec, so certified to run on K8s.

### Architecture

LocalPV refers to storage that is directly attached to a specific node in the Kubernetes cluster. It uses locally available disks (e.g., SSDs, HDDs) on the node.

<b>Use Case</b>: Ideal for workloads that require low-latency access to storage or when data locality is critical (e.g., databases, caching systems).

> #### Characteristics:

Reviewer 2: ```suggestion
#### Characteristics:

```

## Overview

### What is OpenEBS ZFS LocalPV?

OpenEBS ZFS LocalPV is a [CSI](https://github.com/container-storage-interface/spec) plugin for implementation of [ZFS](https://en.wikipedia.org/wiki/ZFS) backed persistent volumes for Kubernetes. It is a local storage solution, which means the device, volume and the application are on the same host. It doesn't contain any dataplane, i.e only its simply a control-plane for the kernel zfs volumes. It mainly comprises of two components which are implemented in accordance to the CSI Specs:

1. CSI Controller - Frontends the incoming requests and initiates the operation.
2. CSI Node Plugin - Serves the requests by performing the operations and making the volume available for the initiator.

### Why OpenEBS ZFS LocalPV?

1. Lightweight, easy to set up storage provisoner for host-local volumes in k8s ecosystem.
2. Makes ZFS stack available to K8s, allowing end users to use the ZFS functionalites like snapshot, restore, clone, thin provisioning, resize, encryption, compression, dedup, etc for their Persistent Volumes.

> 3. Cloud native, i.e based on CSI spec, so certified to run on K8s.

Reviewer 2: Maybe certified not a good term here, since there's no certification?

Author: Sure.

Author: `3. Cloud native, i.e based on CSI spec, so developed to run on K8s.`

How about this?

    - [x] `Block` mode
- [x] Supports fsTypes: `ext4`, `btrfs`, `xfs`, `zfs`
- [x] Volume metrics
- [x] [Snapshot]
    - [x] [Create](docs/snapshot.md)
    - [x] [Restore](docs/clone.md#create-clone-from-snapshot)
- [x] [Clone](docs/clone.md#create-clone-from-volume)
- [x] [Volume Resize](docs/resize.md)
- [x] [Raw Block Volume](docs/raw-block-volume.md)
- [x] [Backup/Restore](docs/backup-restore.md)
- [ ] Ephemeral inline volume

> ## Dev Activity dashboard

Reviewer 2: ```suggestion
## Dev Activity dashboard

```
